### ISSUE TITLE
View::registerJsFile() registers the same file multiple times

### ISSUE THREAD

**Issue by PowerGamer1** [yiisoft/yii2#10172]
As per `\yii\web\View::registerJsFile()` documentation:
```
     * @param string $key the key that identifies the JS script file. If null, it will use
     * $url as the key. If two JS files are registered with the same key, the latter
     * will overwrite the former.
```
The following code behaves correctly (as per documentation) - the file 1.js is registered only once:
```
$view->registerJsFile('c:\dir\1.js', ['position' => \yii\web\View::POS_HEAD, 'depends' => [\frontend\assets\AppAsset::className()]]);
$view->registerJsFile('c:\dir\1.js', ['position' => \yii\web\View::POS_END, 'depends' => [\frontend\assets\AppAsset::className()]]);
```
The following code behaves incorrectly - the file 1.js is registered twice in the HTML document:
```
$view->registerJsFile('c:\dir\1.js', ['position' => \yii\web\View::POS_HEAD]);
$view->registerJsFile('c:\dir\1.js', ['position' => \yii\web\View::POS_END]);
```

**Comment by Alex-Code** [comment]
The internal array is indexed by the position first so having having `POS_HEAD` and `POS_END` would allow this.

Why are you registering the same script in different positions anyway?

**Comment by PowerGamer1** [comment]
> The internal array is indexed by the position first so having having  POS_HEAD  and  POS_END  would allow this.

But if the file is registered with dependency the behavior of ` \yii\web\View::registerJsFile() ` would be different, so its a bug in ` \yii\web\View::registerJsFile() `.

> Why are you registering the same script in different positions anyway?

For example, it can happen when the registration is done from different places in the code.

**Comment by resurtm** [comment]
Looks like it should be fixed to stick the documentation. I also see no issues related to BC. Will work on this.

**Comment by klimov-paul** [comment]
Resolved by commit c41ced6f1a164fe51b155fbef2ae12b084b08fac
